src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,25 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  books: [],
+};
+
+const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.books.push(action.payload);
+    },
+    removeBook: (state, action) => {
+      const bookIndex = state.books.findIndex((book) => book.id === action.payload);
+      if (bookIndex !== -1) {
+        state.books.splice(bookIndex, 1);
+      }
+    },
```

Comment:
- [ ] **[OPTIONAL]** In React it is not a good practice to mutate the state, rather we should make a copy of the array. The `splice()` method mutates the array. Kindly refactor your method so that it makes a copy of the array, instead of changing it directly. Here is an example below - note that the `filter()` method does not mutate the state, it rather makes a copy.

  ```
  removeBook: (state, action) => {
    state.books = state.books.filter((book) => book.id !== action.id)
  },
  ```

Reply:
I will implement it, thank you!